code/modules/xenoarchaeology/traits/traits.dm

```diff
@@ -0,0 +1,260 @@
+	var/incompatabilities
+
+/datum/xenoartifact_trait/New(atom/_parent)
+	. = ..()
+	if(_parent)
+		register_parent(_parent)
+
+/datum/xenoartifact_trait/Destroy(force, ...)
+	. = ..()
+	dump_targets()
+	remove_parent(parent, FALSE)
+
+//The reason this is a seperate proc is so we can init the trait and swap its artifact component parent around
+/datum/xenoartifact_trait/proc/register_parent(datum/source)
+	parent = source
+	var/atom/movable/AM = parent.parent
+	RegisterSignal(parent, COMSIG_PARENT_QDELETING, PROC_REF(remove_parent))
+	//Setup trigger signals
+	RegisterSignal(parent, COMSIG_XENOA_TRIGGER, PROC_REF(trigger))
+	//If we need to setup signals for pearl stuff
+	if(can_pearl)
+		RegisterSignal(AM, COMSIG_ATOM_TOOL_ACT(TOOL_SCREWDRIVER), PROC_REF(catch_pearl_tool))
+		RegisterSignal(AM, COMSIG_MOVABLE_MOVED, PROC_REF(catch_move))
+	//Appearance
+	if(parent.do_texture)
+		generate_trait_appearance(parent.parent)
+	//Stats
+	parent.target_range += extra_target_range
+	AM.custom_price += extra_value
+
+//Remeber to call this before setting a new parent
+/datum/xenoartifact_trait/proc/remove_parent(datum/source, pensive = TRUE)
+	SIGNAL_HANDLER
+
+	//Detach from current parent
+	if(parent)
+		remove_hints()
+		UnregisterSignal(parent, COMSIG_PARENT_QDELETING)
+		UnregisterSignal(parent, COMSIG_XENOA_TRIGGER)
+		var/atom/A = parent.parent
+		parent.target_range -= extra_target_range
+		A.custom_price -= extra_value
+		cut_trait_appearance(parent.parent)
+		if(can_pearl)
+			UnregisterSignal(A, COMSIG_ATOM_TOOL_ACT(TOOL_SCREWDRIVER))
+			UnregisterSignal(A, COMSIG_MOVABLE_MOVED)
+	if(pensive)
+		qdel(src)
+	parent = null
+	dump_targets()
+
+//Cleanly register an effected target
+/datum/xenoartifact_trait/proc/register_target(atom/target, do_trigger = FALSE)
+	if(do_trigger)
+		trigger(null, priority, target)
+	targets += target
+	RegisterSignal(target, COMSIG_PARENT_QDELETING, PROC_REF(unregister_target_signal), TRUE)
+
+//Cleanly unregister an effected target
+/datum/xenoartifact_trait/proc/unregister_target(datum/source, do_untrigger = FALSE)
+	SIGNAL_HANDLER
+
+	if(do_untrigger) //This will only happen in the event something is unregistered before we can untrigger, which is needed for QDELs
+		un_trigger(source, override = source)
+	targets -= source
+	UnregisterSignal(source, COMSIG_PARENT_QDELETING)
+
+/datum/xenoartifact_trait/proc/unregister_target_signal(datum/source)
+	SIGNAL_HANDLER
+
+	unregister_target(source, TRUE)
+
+/datum/xenoartifact_trait/proc/trigger(datum/source, _priority, atom/override)
+	SIGNAL_HANDLER
+
+	. = TRUE
+	if(_priority != priority && _priority)
+		return FALSE
+	if(!register_targets)
+		return
+	//If we've been given an override
+	if(override)
+		register_target(override)
+	//Otherwise just use the artifact's target list
+	else if(length(parent.targets))
+		for(var/atom/I in parent.targets)
+			register_target(I)
+	//Handle focus
+	focus = override ? list(override) : targets
```

Comment:
does it need to be wrapped with `list()` if it exists already?
did you want `override.Copy()`?

Reply:
override is a single atom we're focusing on, for whatever reason